BridgeAppSDK/ORKResult+SBAExtension.swift

```diff
@@ -0,0 +1,341 @@
+        }
+        return  rectangles
+    }
+}
+
+class AnswerKeyAndValue: NSObject {
+    var key: String
+    var value: AnyObject?
+    
+    init(key: String, value: AnyObject?) {
+        self.key = key
+        self.value = value
+    }
+}
+
+protocol ORKQuestionResultAnswerJSON {
+    func jsonSerializedAnswer() -> (AnswerKeyAndValue)?
+}
+
+private var _timeOnlyFormatter: NSDateFormatter? = nil
+
+extension ORKQuestionResult: ORKQuestionResultAnswerJSON {
+    
+    func timeOnlyFormatter() -> NSDateFormatter {
+        if _timeOnlyFormatter == nil {
+            _timeOnlyFormatter = NSDateFormatter.init()
+            _timeOnlyFormatter!.dateFormat = "HH:mm:ss.SSS"
+            let enUSPOSIXLocale = NSLocale.init(localeIdentifier: "en_US_POSIX");
+            _timeOnlyFormatter!.locale = enUSPOSIXLocale
+        }
+        
+        return _timeOnlyFormatter!
+    }
+    override func bridgeData(stepIdentifier: String) -> ArchiveableResult? {
+        let filename = bridgifyFilename(self.identifier) + ".json"
+        return ArchiveableResult.init(result: self.jsonObject(), resultType: .Dictionary, filename: filename)
+    }
+    
+    public override func jsonObject() -> AnyObject {
+        var ChoiceQuestionResult = [String: AnyObject]()
+        ChoiceQuestionResult[QuestionResultQuestionTextKey] = self.questionText
+        ChoiceQuestionResult[QuestionResultQuestionTypeKey] = self.questionType.rawValue
+        ChoiceQuestionResult[kIdentifierKey] = self.identifier
+        ChoiceQuestionResult[kStartDateKey] = self.startDate
+        ChoiceQuestionResult[kEndDateKey] = self.endDate
+        ChoiceQuestionResult[QuestionResultUserInfoKey] = self.userInfo
+        if let answer = self.jsonSerializedAnswer() {
+            ChoiceQuestionResult[answer.key] = answer.value
+        }
+        return (ChoiceQuestionResult as NSDictionary).jsonObject()
+    }
+    
+    func jsonSerializedAnswer() -> (AnswerKeyAndValue)? {
+        let className = NSStringFromClass(self.classForCoder)
+        fatalError("jsonSerializedAnswer not implemented for \(className)")
+    }
+    
+    func jsonForAnswer(answer: AnyObject) -> AnyObject {
+        let dateTime = answer as? NSDate
+        switch self.questionType {
+        case .TimeOfDay:
+            return timeOnlyFormatter().stringFromDate(dateTime!)
+        case .DateAndTime:
+            return dateTime!.ISO8601String()
+        case .Date:
+            return dateTime!.ISO8601DateOnlyString()
+        default:
+            // TODO: Implement specific formatters for other questionTypes as needed
+            return answer.jsonObject()
+        }
+    }
+}
+
+extension ORKChoiceQuestionResult {
+    override func jsonSerializedAnswer() -> (AnswerKeyAndValue)? {
+        let choiceAnswers = self.choiceAnswers?.map { (answer) -> AnyObject in
+            return jsonForAnswer(answer)
+        }
+        return AnswerKeyAndValue.init(key: "choiceAnswers", value: choiceAnswers)
```

Comment:
````
guard let choiceAnswers = self.choiceAnswers?.map( { jsonForAnswer($0) }) else { return nil }
return AnswerKeyAndValue(key: "choiceAnswers", value: choiceAnswers)
````
Also, I think that surveys might be designed to look at "answer" for all values, rather than the property associated with the specific type. Should probably check that.